src/Sylius/Behat/Context/Api/Admin/ManagingShippingCategoriesContext.php

```diff
@@ -0,0 +1,223 @@
+        }
+    }
+
+    /**
+     * @When I name it :shippingCategoryName
+     * @When I do not specify its name
+     */
+    public function iNameIt(?string $shippingCategoryName = null): void
+    {
+        if ($shippingCategoryName !== null) {
+            $this->client->addRequestData('name', $shippingCategoryName);
+        }
+    }
+
+    /**
+     * @When /^I modify a (shipping category "([^"]+)")$/
+     */
+    public function iModifyAShippingCategory(ShippingCategoryInterface $shippingCategory): void
+    {
+        $this->client->buildUpdateRequest('shipping_categories', (string) $shippingCategory->getId());
+    }
+
+    /**
+     * @When I rename it to :name
+     */
+    public function iNameItIn(string $name): void
+    {
+        $this->client->addRequestData('name', $name);
+    }
+
+    /**
+     * @When I save my changes
+     */
+    public function iSaveMyChanges(): void
+    {
+        $this->client->update();
+    }
+
+    /**
+     * @When I specify its description as :shippingCategoryDescription
+     */
+    public function iSpecifyItsDescriptionAs(string $shippingCategoryDescription): void
+    {
+        $this->client->addRequestData('description', $shippingCategoryDescription);
+    }
+
+    /**
+     * @Then I should be notified that shipping category with this code already exists
+     */
+    public function iShouldBeNotifiedThatShippingCategoryWithThisCodeAlreadyExists(): void
+    {
+        Assert::same($this->client->getError(), 'code: The shipping category with given code already exists.');
+    }
+
+    /**
+     * @Then I should be notified that :element is required
+     */
+    public function iShouldBeNotifiedThatCodeIsRequired(string $element): void
+    {
+        Assert::same(
+            $this->client->getError(), sprintf('%s: Please enter shipping category %s.', $element, $element)
+        );
+    }
+
+    /**
+     * @Then I should see a single shipping category in the list
+     * @Then I should see :numberOfShippingCategories shipping categories in the list
+     */
+    public function iShouldSeeShippingCategoriesInTheList(int $numberOfShippingCategories = 1): void
+    {
+        $this->client->index('shipping_categories');
+        Assert::same($this->client->countCollectionItems(), $numberOfShippingCategories);
+    }
+
+    /**
+     * @Then /^the (shipping category "([^"]+)") should be in the registry$/
+     * @Then /^the (shipping category "([^"]+)") should appear in the registry$/
+     */
+    public function theShippingCategoryShouldAppearInTheRegistry(ShippingCategoryInterface $shippingCategory): void
+    {
+        $this->client->index('shipping_categories');
+        $shippingCategoryName = $shippingCategory->getName();
+        Assert::true(
+            $this->client->hasItemWithValue('name', $shippingCategoryName),
+            sprintf('Shipping category with name %s does not exist', $shippingCategoryName)
+        );
```

Comment:
You used this code 3 times, i suggest to extract it to private method.

Reply:
I merged one, the rest that left is different, has false,true or different methods inside like hasItemWithValue() or hasValue

Reply:
I don't insist but:
```
private function isItemOnIndex(string $property, string $value): bool
{
        $this->client->index('shipping_category');
        return $this->client->hasItemWithValue($property, $value);
    }
```
and you can assert true/false :)

Reply:
Did you try Adam's proposition? :)